src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1666,15 +1637,9 @@ private Http2Stream AddStream(HttpRequestMessage request)
         {
             lock (SyncObject)
             {
-                if (_nextStream == MaxStreamId || _disposed || _lastValidStreamId != -1)
+                if (_nextStream == MaxStreamId || _disposed || _lastStreamId != -1)
                 {
-                    if (_abortException != null)
-                    {
-                        // Aborted because protocol error or IO.
-                        throw new ObjectDisposedException(nameof(Http2Connection), _abortException);
-                    }
-
-                    // We run out of IDs or we have race condition between receiving GOAWAY and processing requests.
+                    // We ran out of stream IDs or we raced condition between acquiring the connection from the pool and shutting down.
```

Comment:
Nit: "or we raced condition"... strange phrasing

Reply:
Will fix